### PR TITLE
[docs] Clarify application_packages

### PR DESCRIPTION
**Pull request by bmorelli25**
Closes: https://github.com/elastic/apm-agent-java/issues/726.

Adds more detail and an example for `application_packages`. Thanks to Felix's discuss response for clarifying things.

I still need to sit down with one of you and discuss how the Java Agent asciidoc files are generated. Do I need to update this in `StacktraceConfiguration.java` and run something to update this asciidoc file? Somewhere else?

### CODE REVIEW

**Comment by felixbarny** [comment]
Yes, this has to be updated in `StacktraceConfiguration.java`. The asciidoc is auto-generated. You can generate it via `./mvnw clean test`.

File: docs/configuration.asciidoc [hunk 1]
```diff
@@ -1097,12 +1097,14 @@ Prepending an element with `(?-i)` makes the matching case sensitive.
...
-In order to be able to use the API annotations @CaptureTransaction and @CaptureSpan,
-it is required to set these options.
+This allows the UI to collapse the stack frames of library code,
+and highlight the stack frames that originate from your application.
+Multiple root packages can be set as a comma-separated list;
+there's no need to configure sub-packages.
+Because this setting helps determine which classes to scan on startup,
+setting this option can also improve startup time.
 
+You must set this option in order to use the API annotations @CaptureTransaction and @CaptureSpan.
```
Comment (felixbarny) [inline docs/configuration.asciidoc]:
```suggestion
You must set this option in order to use the API annotations `@CaptureTransaction` and `@CaptureSpan`.
```

**Comment by bmorelli25** [comment]
Thanks Felix! I updated `StacktraceConfiguration.java`.

File: apm-agent-core/src/main/java/co/elastic/apm/agent/impl/stacktrace/StacktraceConfiguration.java [hunk 2]
```diff
@@ -40,11 +40,24 @@
...
+            "\n" +
+            "**Example**\n" +
+            "\n" +
+            "Most Java projects have a root package, e.g. `com.myproject`.\n" +
+            "Here's how you can set that application package using Java system properties:\n" +
+            "\n" +
+            "[source,bash]\n" +
+            "----\n" +
+            "-Delastic.apm.application_packages=com.myproject\n" +
+            "----\n")
```
Comment (eyalkoren) [inline apm-agent-core/src/main/java/co/elastic/apm/agent/impl/stacktrace/StacktraceConfiguration.java]: The problem is that this goes also to the elasticapm.properties file, where it doesn't make sense.
I'd say get rid of these HTML hints. I think using backticks for that would be enough, as in:
`-Delastic.apm.application_packages=com.myproject`.

Comment (eyalkoren) [inline apm-agent-core/src/main/java/co/elastic/apm/agent/impl/stacktrace/StacktraceConfiguration.java]: For additional clarity, I would add something like:
```suggestion
            "And if you are only interested in specific subpackages, you can set something like:\n" + 
            `-Delastic.apm.application_packages=com.myproject.api,com.myproject.impl`\n")
```

**Comment by bmorelli25** [comment]
Thanks for the fix @eyalkoren. I've update this PR per your suggestions.

**Pull request merged**